### PR TITLE
processor,sink(cdc): let sink report resolved ts and do not skip buffer sink flush

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Revert #3029 and #2617 temporary for resolving #3503.

Cc https://github.com/pingcap/ticdc/issues/3545

### Check List 

Tests 

 - Integration test

See #3503

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/owner/changefeed.go [hunk 1]
```diff
@@ -391,6 +391,10 @@ func (c *changefeed) handleBarrier(ctx cdcContext.Context) (uint64, error) {
 		if !blocked {
 			return barrierTs, nil
 		}
+		log.Info("advance DDL barrier",
```
Comment (liuzix) [inline cdc/owner/changefeed.go:394]: The message here can be confusing to users and support staff. The "DDL barrier" does not seem to be actually changed here. Also, for a single DDL, this piece of code can be executed more than once, causing more confusion.

File: cdc/sink/table_sink.go [hunk 2]
```diff
@@ -55,6 +57,15 @@ func (t *tableSink) EmitDDLEvent(ctx context.Context, ddl *model.DDLEvent) error
 // is required to be no more than global resolvedTs, table barrierTs and table
 // redo log watermarkTs.
 func (t *tableSink) FlushRowChangedEvents(ctx context.Context, resolvedTs uint64) (uint64, error) {
+	logAbnormalCheckpoint := func(ckpt uint64) {
```
Comment (liuzix) [inline cdc/sink/table_sink.go:60]: Can you add some comments here?

File: cdc/sink/table_sink.go [hunk 3]
```diff
@@ -55,6 +57,15 @@ func (t *tableSink) EmitDDLEvent(ctx context.Context, ddl *model.DDLEvent) error
...
+		if ckpt > resolvedTs {
+			log.L().WithOptions(zap.AddCallerSkip(1)).
+				Warn("checkpoint ts > resolved ts, flushed more than emitted",
```
Comment (amyangfei) [inline cdc/sink/table_sink.go:63]: If this happens, is it better to return an error and pause the changefeed

**Review by amyangfei — APPROVED** [review]

File: cdc/sink/table_sink.go (hunk 3 above)
Comment (overvenus) [reply cdc/sink/table_sink.go:64]: The probability is high after table rescheduling, return error may cause changefeed stuck easily. 
In addition, this situation does not necessarily mean data loss.

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**